internal/auth0/role/resource_permission_test.go

```diff
@@ -0,0 +1,129 @@
+resource "auth0_role_permission" "another_role_permission" {
+	depends_on = [ auth0_role_permission.another_role_permission ]
+
+	role_id = auth0_role.role.id
+	resource_server_identifier = auth0_resource_server.resource_server.identifier
+	permission = "read:foo"
+}
+
+data "auth0_role" "role" {
+	depends_on = [ auth0_role_permission.role_permission, auth0_role_permission.another_role_permission ]
+	role_id = auth0_role.role.id
+}
+`
+
+func TestAccRolePermission(t *testing.T) {
+	acctest.Test(t, resource.TestCase{
+		Steps: []resource.TestStep{
+			{
+				Config: acctest.ParseTestName(testAccRolePermissionsNoneAssigned, strings.ToLower(t.Name())),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("data.auth0_role.role", "permissions.#", "0"),
+				),
+			},
+			{
+				Config: acctest.ParseTestName(testAccRolePermissionsOneAssigned, strings.ToLower(t.Name())),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("data.auth0_role.role", "permissions.#", "1"),
+					resource.TestCheckResourceAttr("data.auth0_role.role", "permissions.0.name", "create:foo"),
+					resource.TestCheckResourceAttr("data.auth0_role.role", "permissions.0.resource_server_identifier", "https://uat.testaccrolepermission.terraform-provider-auth0.com/api"),
+
+					resource.TestCheckResourceAttr("auth0_role_permission.role_permission", "permission", "create:foo"),
+					resource.TestCheckResourceAttr("auth0_role_permission.role_permission", "resource_server_identifier", "https://uat.testaccrolepermission.terraform-provider-auth0.com/api"),
+					resource.TestCheckResourceAttr("auth0_role_permission.role_permission", "resource_server_name", "Acceptance Test - testaccrolepermission"),
+					resource.TestCheckResourceAttr("auth0_role_permission.role_permission", "description", "Can create Foo"),
+				),
+			},
+			{
+				Config: acctest.ParseTestName(testAccRolePermissionsTwoAssigned, strings.ToLower(t.Name())),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("auth0_role_permission.role_permission", "permission", "create:foo"),
+					resource.TestCheckResourceAttr("auth0_role_permission.another_role_permission", "permission", "read:foo"),
+
+					resource.TestCheckResourceAttr("data.auth0_role.role", "permissions.#", "2"),
+					resource.TestCheckResourceAttr("data.auth0_role.role", "permissions.0.name", "create:foo"),
+					resource.TestCheckResourceAttr("data.auth0_role.role", "permissions.1.name", "read:foo"),
+
+					resource.TestCheckResourceAttr("data.auth0_role.role", "permissions.0.name", "create:foo"),
+					resource.TestCheckResourceAttr("data.auth0_role.role", "permissions.0.resource_server_identifier", "https://uat.testaccrolepermission.terraform-provider-auth0.com/api"),
+				),
+			},
+			{
+				Config: acctest.ParseTestName(testAccRolePermissionsOneAssigned, strings.ToLower(t.Name())),
+			},
+			{
+				RefreshState: true,
```

Comment:
```suggestion
				RefreshState: true,
```

![image](https://github.com/auth0/terraform-provider-auth0/assets/28300158/0a303f34-e44d-4dcb-9319-c16f2b2950ee)

Should this be here? I think we can move the checks in the config above if we use the data source as the last elements that gets refreshed.

Reply:
This needs to be here for tests to pass. The data source gets executed after (or at best, during) the deletions of the `auth0_role_permission` resources. AFAICT, there is no way to depend on a deleted resource.

Reply:
Ah, I see that now 👍🏻 , indeed the execution flow on a terraform apply is:

read and refresh state
creations
updates
deletions

so we need another refresh state after the deletion.